pkg/ccip/bindings/common/common.go

```diff
@@ -419,3 +419,33 @@ func NewDummyCell() (*cell.Cell, error) {
 	}
 	return builder.EndCell(), nil
 }
+
+// Proof256 represents a 32-byte (256 bits) proof used in merkle proofs.
+// This wrapper type allows [32]byte to be used with SnakeData by implementing
+// ToCell/LoadFromCell that directly store/load 256 bits inline, avoiding the
+// infinite loop issue that occurs with SnakeBytes (which uses c.ToCell() in LoadFromCell).
+type Proof256 [32]byte
```

Comment:
Could we simplified this wrapped type by using a wrapped type with a standard tlb tag like so: https://github.com/smartcontractkit/chainlink-ton/blob/main/pkg/bindings/mcms/mcms/types.go#L228

```
type Proof struct {
	Value *big.Int `tlb:"## 256"` // The value of the struct
}

// Usage example
MetadataProof common.SnakeData[Proof]                `tlb:"^"` // The MerkleProof of inclusion of the metadata in the Merkle tree. // vec<uint256>
```